src/files/object.js

```diff
@@ -113,6 +112,14 @@ class ObjectFile extends BaseFile {
       formData.append('file', this.__file, this.fileName)
       formData.append('file_name', this.fileName)
       formData.append('source', this.sourceInfo.source)
+
+      const metadata = getMetadataObject(this.settings)
+      if (metadata) {
+        $.each(metadata, (key, value) => {
+          formData.append(`metadata[${key}]`, String(value))
+        })
+      }
```

Comment:
Check if possible to simplify API:

```python
for k, v in getMetadataObject(this.settings):
    data[`metadata[${k}]`] = String(v)

for k, v in getMetadataObject(this.settings):
    formData.append[`metadata[${k}]`] = String(v)
```

Reply:
We can't use `for of` and `Object.entries` due to compatibility. So the only way is to use `$.each` to iterate over `k, v`.

I removed `if` condition:

```js
const metadata = getMetadataObject(this.settings)
$.each(metadata, (key, value) => {
  data[`metadata[${key}]`] = String(value)
})
```

It's also possible to reduce lines like this:

```js
$.each(getMetadataObject(this.settings), (key, value) => {
  data[`metadata[${key}]`] = String(value)
})
```

but it looks less clean for me.

Reply:
Moved all string casting and iteration logic under the `utils`. 

```js
iterateMetadata(this.settings, (key, value) => {
  data[`metadata[${key}]`] = value
})
```

Reply:
I think this is the best option:

```js
$.each(getMetadataObject(this.settings), (key, value) => {
  data[`metadata[${key}]`] = String(value)
})
```

Reply:
Yeah, you're right. But I would move type casting to the `getMetadataObject`.

```js
      $.each(getMetadataObject(this.settings), (key, value) => {
        formData.append(`metadata[${key}]`, value)
      })
```